types/defines/images.d.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2024 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+type InfoResponse =
+  | { format: "image/svg+xml" }
+  | {
+      format: string;
+      fileSize: number;
+      width: number;
+      height: number;
+    };
+
+type RawInfoResponse =
+  | { format: "image/svg+xml" }
+  | {
+      format: string;
+      file_size: number;
+      width: number;
+      height: number;
+    };
+
+type Transform = {
+  fit?: "scale-down" | "contain" | "pad" | "squeeze" | "cover" | "crop";
+  gravity?:
+    | "left"
+    | "right"
+    | "top"
+    | "bottom"
+    | "center"
+    | "auto"
+    | "entropy"
+    | "face"
+    | {
+        x?: number;
+        y?: number;
+        mode: "remainder" | "box-center";
+      };
+  trim?: {
+    top?: number;
+    bottom?: number;
+    left?: number;
+    right?: number;
+    width?: number;
+    height?: number;
+    border?:
+      | boolean
+      | {
+          color?: string;
+          tolerance?: number;
+          keep?: number;
+        };
+  };
+  width?: number;
+  height?: number;
+  background?: string;
+  rotate?: number;
+  sharpen?: number;
+  blur?: number;
+  contrast?: number;
+  brightness?: number;
+  gamma?: number;
+  border?: {
+    color?: string;
+    width?: number;
+    top?: number;
+    bottom?: number;
+    left?: number;
+    right?: number;
+  };
+  zoom?: number;
+};
+
+type OutputOptions = {
+  format:
+    | "image/jpeg"
+    | "image/png"
+    | "image/gif"
+    | "image/webp"
+    | "image/avif"
+    | "rgb"
+    | "rgba";
+  quality?: number;
+  background?: string;
+};
+
+interface ImagesBinding {
+  info(blob: Blob): Promise<InfoResponse>;
+  input(blob: Blob): ImageTransformer;
+}
+
+interface ImageTransformer {
+  transform(transform: Transform): ImageTransformer;
+  output(options: OutputOptions): Promise<TransformationResult>;
+}
+
+interface TransformationResult {
+  response(): Response;
+  contentType(): string;
+  image(): ReadableStream<Uint8Array> | null;
+}
+
+interface ImagesError {
+  code: number;
```

Comment:
I wonder if there is a precedence for error handling in bindings. It seems that R2 and D1 as preexisting APIs take vastly different approach to this.

Could we think about which way we want to go in this area, and start consolidating the error APIs to feel consistent?

Reply:
at minimum this property should be readonly. For reference this is what R2Error looks like:

```

declare interface R2Error extends Error {
  readonly name: string;
  readonly code: number;
  readonly message: string;
  readonly action: string;
  readonly stack: any;
}
```
source: https://unpkg.com/browse/@cloudflare/workers-types@4.20240614.0/experimental/index.d.ts

Reply:
Made it this for now:
```ts
interface ImagesError extends Error {
  readonly code: number;
  readonly message: string;
  readonly stack?: string;
}
```

It's fairly similar to the Vectorize error (which I cribbed from shamelessly)